### ISSUE TITLE
NullPointer Exception on Android 12 on Resume

### ISSUE THREAD

**Issue by rogerbriggen** [microsoft/appcenter-sdk-dotnet#1701]
### **Description**

We see crashed of our app because of appcenter sdk

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Currently, I only see the stacktrace on appcenter... but it seems if the app resumes

java.lang.NullPointerException: Attempt to invoke interface method 'boolean android.content.SharedPreferences.getBoolean(java.lang.String, boolean)' on a null object reference
com.microsoft.appcenter.utils.storage.SharedPreferencesManager.getBoolean SharedPreferencesManager.java:68
com.microsoft.appcenter.AbstractAppCenterService.isInstanceEnabled AbstractAppCenterService.java:136
com.microsoft.appcenter.distribute.Distribute.registerReceiver Distribute.java:612
com.microsoft.appcenter.distribute.Distribute.onActivityResumed Distribute.java:550
android.app.Application.dispatchActivityResumed Application.java:430
android.app.Activity.dispatchActivityResumed Activity.java:1400
android.app.Activity.onResume Activity.java:1966
androidx.fragment.app.FragmentActivity.onResume FragmentActivity.java:434
android.app.Instrumentation.callActivityOnResume Instrumentation.java:1491
android.app.Activity.performResume Activity.java:8280
android.app.ActivityThread.performResumeActivity ActivityThread.java:4923
android.app.ActivityThread.handleResumeActivity ActivityThread.java:4972
android.app.servertransaction.ResumeActivityItem.execute ResumeActivityItem.java:60
android.app.servertransaction.ActivityTransactionItem.execute ActivityTransactionItem.java:45
android.app.servertransaction.TransactionExecutor.executeLifecycleState TransactionExecutor.java:176
android.app.servertransaction.TransactionExecutor.execute TransactionExecutor.java:97
android.app.ActivityThread$H.handleMessage ActivityThread.java:2345
android.os.Handler.dispatchMessage Handler.java:106
android.os.Looper.loopOnce Looper.java:233
android.os.Looper.loop Looper.java:344
android.app.ActivityThread.main ActivityThread.java:8210
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:584
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1034

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - MAUI (Android)
2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
    - .Net 6
2. Which SDK version are you using?
    - 5.0.0 Preview 1
4. Which OS version did you experience the issue on?
    - Android 12
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    - different physical phones (OnePlus 8T, Mi 9 SE, Galaxy S10, ...)
6. What third party libraries are you using? 
    - 
7. Please enable verbose logging for your app using `AppCenter.LogLevel = LogLevel.Verbose` before your call to `AppCenter.Start(...)` and include the logs here:

**Comment by aleksandr-dorofeev** [comment]
Hello, @rogerbriggen,  thanks for getting in touch with us!
Unfortunately, we were unable to reproduce the behavior you described.
Could you please provide repro steps and [verbose logs](https://learn.microsoft.com/en-gb/appcenter/sdk/other-apis/xamarin#adjust-the-log-level).

**Comment by rogerbriggen** [comment]
Sure. Where do I get then the output of these logs?

**Comment by rogerbriggen** [comment]
Then I cannot. As I have written, the phones are not connected to my computer. 
It would be helpful if we could get a callback for the log entries or you support ILogger

**Comment by aleksandr-dorofeev** [comment]
Sorry for the misunderstanding.
In this case, could you clarify how you launch the application? Do you use AppCenter tests?

**Comment by rogerbriggen** [comment]
No. We just use Crash, Analytics and Distribute in our app, otherwise we don't use any service of app center.
The app is created with MAUI. Currently, I just see several crashes inside appcenter with the above stacktrace which means, that app center sdk is crashing my app. I also think, that the stack trace is quite clear, so catching the exception or checking for null inside the app center is appreciated. The other option is to disable app center distribute since the stack trace seems quite clear about it.

**Comment by DmitriyKirakosyan** [comment]
Hi @rogerbriggen , this looks like a duplicate of https://github.com/microsoft/appcenter-sdk-android/issues/1610 and https://github.com/microsoft/appcenter-sdk-android/issues/1645. It was fixed in https://github.com/microsoft/appcenter-sdk-android/pull/1622 and released in  [4.5.3](https://github.com/microsoft/appcenter-sdk-dotnet/releases/tag/4.5.3).

Based on your stack trace it looks like the version of AppCenter SDK is prior this fix ( < 4.5.3).
Please check the versions of your app for these crashes. There should not be a version which uses SDK 4.5.3 or higher.

**Comment by rogerbriggen** [comment]
Hi  @DmitriyKirakosyan 
As I have written, we use appcenter sdk 5.0.0 Preview 1... So, this is seems there is a regression

**Comment by MikhailSuendukov** [comment]
I close this issue because of inactivity. Please feel to reopen it, in case problem still persist, or you have any questions.
